src/Analyzers/Core/Analyzers/AbstractCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -64,12 +41,6 @@ protected AbstractCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDescripto
             _localizableMessageFormat = Descriptor.MessageFormat;
         }
 
-        protected DiagnosticDescriptor CreateUnnecessaryDescriptor(string descriptorId, bool isConfigurable = true)
-            => CreateDescriptorWithId(
-                descriptorId, _localizableTitle, _localizableMessageFormat,
-                isUnnecessary: true,
-                isConfigurable);
-
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
 
         protected static DiagnosticDescriptor CreateDescriptorWithId(
```

Comment:
Can we get rid of the `isUnnecessary` parameter and corresponding custom tag?

Reply:
➡️ Cannot be removed due to https://github.com/dotnet/roslyn/pull/46666#discussion_r470874219